modules/core/shared/src/main/scala/io/circe/Json.scala

```diff
@@ -537,7 +543,7 @@ object Json {
   /**
    * Create a `Json` value representing a JSON number from an `Int`.
    */
-  final def fromInt(value: Int): Json = JNumber(JsonLong(value.toLong))
+  final def fromInt(value: Int): Json = JNumber(JsonNumber.fromLong(value.toLong))
```

Comment:
If you change the table to `JNumber` as suggested in another commented then you would also change here:

```suggestion
  final def fromInt(value: Int): Json = JsonNumber.fromLong(value.toLong)
```